src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Annotate.java

```diff
@@ -710,16 +709,18 @@ private Attribute getAnnotationArrayValue(Type expectedElementType, JCExpression
         JCNewArray na = (JCNewArray)tree;
         if (na.elemtype != null) {
             log.error(na.elemtype.pos(), Errors.NewNotAllowedInAnnotation);
+            return new Attribute.Error(syms.errType);
+        } else {
+            ListBuffer<Attribute> buf = new ListBuffer<>();
+            for (List<JCExpression> l = na.elems; l.nonEmpty(); l = l.tail) {
+                buf.append(attributeAnnotationValue(types.elemtype(expectedElementType),
```

Comment:
Here, I would be more inclined to attribute the values even in the erroneous case. Like if someone writes:
```
public String[] value() default new String[0] {Other.CONSTANT}
```

Attributing even for the erroneous case would mean the AST would have types/symbols filled in, for possible further use.